src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/diskmanipulator/NetworkNodeDiskManipulator.java

```diff
@@ -195,19 +195,23 @@ private void insertItemIntoNetwork(IStorageDisk<ItemStack> storage, int slot) {
             // We need to check if the stack was inserted
             storage.insert(((extracted == remainder) ? remainder.copy() : remainder), remainder.getCount(), false);
         }
-
-        if (storage.getStacks().size() == 0) {
-            moveDriveToOutput(slot);
-        }
     }
 
     //Iterate through disk stacks, if none can be inserted, return that it is done processing and can be output.
     private boolean checkItemDiskDone(IStorageDisk<ItemStack> storage, int slot) {
-        if (storage.getStored() == 0) {
+        if (ioMode == IO_MODE_INSERT && storage.getStored() == 0) {
             moveDriveToOutput(slot);
             return true;
         }
 
+        //In Extract mode, we just need to check if the disk is full or not.
+        if (ioMode == IO_MODE_EXTRACT)
+            if (storage.getStored() == storage.getCapacity()) {
+                moveDriveToOutput(slot)
```

Comment:
I don't believe you tested this. This line doesn't even compile, that's also why the build is failing. How did you "test" without this line being able to compile?